lib/vanagon/component/source.rb

```diff
@@ -1,78 +1,114 @@
+require 'fustigit'
 require 'vanagon/component/source/http'
 require 'vanagon/component/source/git'
 require 'vanagon/component/source/local'
 
 class Vanagon
   class Component
     class Source
-      SUPPORTED_PROTOCOLS = ['file', 'http', 'git']
-      @@rewrite_rule = {}
+      SUPPORTED_PROTOCOLS = %w(file http https git).freeze
+      @rewrite_rules = {}
 
-      def self.register_rewrite_rule(protocol, rule)
-        if rule.is_a?(String) or rule.is_a?(Proc)
-          if SUPPORTED_PROTOCOLS.include?(protocol)
-            @@rewrite_rule[protocol] = rule
+      class << self
+        attr_reader :rewrite_rules
+
+        # Deprecate this
```

Comment:
Should we add a deprecation warning to it now? Like, please don't build upon this?